cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -51,6 +51,7 @@ type azureCache struct {
 	// Cache content.
 	resourceGroup        string
 	vmType               string
+	vmsPoolMap           map[string]struct{} // track the nodepools that're vms pool
```

Comment:
This `map[string]struct{}` is just being used as set here, since golang doesn't have `set` data structure. 
We don't really set any value to it. The purpose is only to be able to find out if an agentpool is vms pool, given its name, so we don't need to use any specific data type here.

We are setting value like this:
```
vmsPoolMap[to.String(vmPoolName)] = struct{}{}
```

Reply:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/go.mod#L40 api machinery should have sets iirc you can use.

Reply:
Thanks Bryce, the `Set` in apimachinery is also a map, so I don't think it's necessary: 

![image](https://github.com/kubernetes/autoscaler/assets/58485997/1156bc22-aa2a-495b-bbc4-c1e77ad8a83d)

Reply:
All go implementations of set will use map[T]EmptyStruct for zero bytes. 

https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#Set is the lotation of the set package though that works with all types. Set in pkg/fields isn't one i have heard of before.

Reply:
Renamed